datahub/dbmaintenance/tasks.py

```diff
@@ -174,3 +174,103 @@ def _copy_foreign_key_to_m2m_field(
     )
 
     return len(objects_to_create)
+
+
+@shared_task(acks_late=True)
+def copy_export_countries_to_company_export_country_model(
+    status,
+    batch_size=5000,
+):
+    """
+    Task that copies all export countries from Company model to CompanyExportCountry
+    """
+    key_switch = {
+        'future_interest': 'future_interest_countries',
+        'currently_exporting': 'export_to_countries',
+    }
+
+    num_updated = _copy_export_countries(key_switch[status], status)
+
+    # If there are definitely no more rows needing processing, return
+    if num_updated < batch_size:
+        return
+
+    copy_export_countries_to_company_export_country_model.apply_async(
+        kwargs={
+            'batch_size': batch_size,
+            'status': status,
+        },
+
+    )
+
+
+@atomic
+def _copy_export_countries(key, status):
+    """
+    Main logic for copying export companies from Company model to
+    CompanyExportCountry one
+    """
+    export_countries = _get_company_countries(key, status)
+    num_updated = _copy_company_countries(
+        key,
+        export_countries,
+        status,
+    )
+
+    logger.info(
+        f'Company.{key} copied to CompanyExportCountry '
+        f'for {num_updated} Company export countries',
+    )
+
+    return num_updated
+
+
+def _get_company_countries(key, status):
+    company_model = apps.get_model('company', 'Company')
+    company_export_country_model = apps.get_model('company', 'CompanyExportCountry')
+
+    any_company_country_subquery = Exists(
+        company_export_country_model.objects.filter(
+            **{
+                'company_id': OuterRef('pk'),
+                'status': status,
+            },
+        ),
+    )
+
+    batch_queryset = company_model.objects.select_for_update().annotate(**{
+        'has_' + key: any_company_country_subquery,
```

Comment:
Would be good to use f-strings instead of concatenation here and below:

```suggestion
        f'has_{key}': any_company_country_subquery,
```